src/core/crypto/ed25519.ts

```diff
@@ -118,6 +135,10 @@ export class Ed25519PublicKey extends AccountPublicKey {
   static isPublicKey(publicKey: AccountPublicKey): publicKey is Ed25519PublicKey {
     return publicKey instanceof Ed25519PublicKey;
   }
+
+  static isInstance(publicKey: PublicKey): publicKey is Ed25519PublicKey {
```

Comment:
could we add tests to all those new `isInstance` functions?

Reply:
done